pkgs/by-name/da/dawarich/package.nix

```diff
@@ -0,0 +1,165 @@
+{
+  lib,
+  applyPatches,
+  bundlerEnv,
+  fetchFromGitHub,
+  fetchNpmDeps,
+  h3_3,
+  nixosTests,
+  nodejs,
+  npmHooks,
+  ruby_3_4,
+  stdenv,
+  tailwindcss_3,
+
+  sources ? ./. + "/sources.json",
+  gemset ? import (./. + "/gemset.nix"),
+}:
+let
+  ruby = ruby_3_4;
+  sources' = lib.importJSON sources;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dawarich";
+  inherit (sources') version;
+
+  src = applyPatches {
+    src = fetchFromGitHub {
+      owner = "Freika";
+      repo = "dawarich";
+      tag = finalAttrs.version;
+      inherit (sources') hash;
+
+    };
+    patches = [
+      # bundix and bundlerEnv fail with system-specific gems
+      ./0001-build-ffi-gem.diff
+      # openssl 3.6.0 breaks ruby openssl gem
+      # See https://github.com/NixOS/nixpkgs/issues/456753
+      # and https://github.com/ruby/openssl/issues/949#issuecomment-3370358680
+      ./0002-openssl-hotfix.diff
+    ];
+    postPatch = ''
+      substituteInPlace ./Gemfile \
+        --replace-fail "ruby File.read('.ruby-version').strip" "ruby '>= 3.4.0'"
+    '';
+  };
```

Comment:
Why not just use `patches`? This would be harder to override so if there's a technical reason to needing this, might be good to have `unpatchedSource` be `src.src` so overrides would be easier to apply.

Reply:
I'm meaning option 2 so someone can change `unpatchedSource` instead of `src` and they wouldn't have to recreate `applyPatches`.